x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/stats/SearchContextStats.java

```diff
@@ -0,0 +1,357 @@
+                } else {
+                    hasIdenticalDelegate = false;
+                }
+            } else {
+                indexed = false;
+                hasDocValues = false;
+                hasIdenticalDelegate = false;
+            }
+        }
+        if (exists == false) {
+            // if it does not exist on any context, no other settings are valid
+            return new FieldConfig(false, false, false, false);
+        } else {
+            return new FieldConfig(exists, hasIdenticalDelegate, indexed, hasDocValues);
+        }
+    }
+
+    public boolean isIndexed(String field) {
+        var stat = cache.computeIfAbsent(field, this::makeFieldStats);
+        return stat.config.indexed;
+    }
+
+    public boolean hasDocValues(String field) {
+        var stat = cache.computeIfAbsent(field, this::makeFieldStats);
+        return stat.config.hasDocValues;
+    }
+
+    public boolean hasIdenticalDelegate(String field) {
+        var stat = cache.computeIfAbsent(field, this::makeFieldStats);
+        return stat.config.hasIdenticalDelegate;
+    }
+
+    public long count() {
+        var count = new long[] { 0 };
+        boolean completed = doWithContexts(r -> {
+            count[0] += r.numDocs();
+            return true;
+        }, false);
+        return completed ? count[0] : -1;
+    }
+
+    public long count(String field) {
+        var stat = cache.computeIfAbsent(field, this::makeFieldStats);
+        if (stat.count == null) {
+            var count = new long[] { 0 };
+            boolean completed = doWithContexts(r -> {
+                count[0] += countEntries(r, field);
+                return true;
+            }, false);
+            stat.count = completed ? count[0] : -1;
+        }
+        return stat.count;
+    }
+
+    public long count(String field, BytesRef value) {
+        var count = new long[] { 0 };
+        Term term = new Term(field, value);
+        boolean completed = doWithContexts(r -> {
+            count[0] += r.docFreq(term);
+            return true;
+        }, false);
+        return completed ? count[0] : -1;
+    }
+
+    public byte[] min(String field, DataType dataType) {
+        var stat = cache.computeIfAbsent(field, this::makeFieldStats);
+        if (stat.min == null) {
+            var min = new byte[][] { null };
+            doWithContexts(r -> {
+                byte[] localMin = PointValues.getMinPackedValue(r, field);
+                // TODO: how to compare with the previous min
+                if (localMin != null) {
+                    if (min[0] == null) {
+                        min[0] = localMin;
+                    } else {
+                        throw new EsqlIllegalArgumentException("Don't know how to compare with previous min");
+                    }
+                }
+                return true;
+            }, true);
+            stat.min = min[0];
+        }
+        // return stat.min;
+        return null;
+    }
```

Comment:
Could Ignacio help finalize these two methods (min and max)?
The min and max come handy with range queries (e.g. `WHERE x < 10`  gets folded to false if x min is 10+).

Reply:
The `min` and `max` methods were coded in the original `SearchStats`, but as yet have never been used. I actually favour removing them, and only adding them back when they get used. A feature to fold `WHERE x < 10` to `false` on a per-shard or per collections of shards basis is certainly a nice feature, and I will discuss this with Ignacio.